packages/remix/src/client/browserTracingIntegration.ts

```diff
@@ -0,0 +1,41 @@
+import { browserTracingIntegration as originalBrowserTracingIntegration } from '@sentry/react';
+import type { Integration } from '@sentry/types';
+import { setGlobals, startPageloadSpan } from './performance';
+import type { RemixBrowserTracingIntegrationOptions } from './performance';
+/**
+ * Creates a browser tracing integration for Remix applications.
+ * This integration will create pageload and navigation spans.
+ */
+export function browserTracingIntegration(options: RemixBrowserTracingIntegrationOptions): Integration {
+  if (options.instrumentPageLoad === undefined) {
+    options.instrumentPageLoad = true;
+  }
+
+  if (options.instrumentNavigation === undefined) {
+    options.instrumentNavigation = true;
+  }
+
+  setGlobals({
+    useEffect: options.useEffect,
+    useLocation: options.useLocation,
+    useMatches: options.useMatches,
+    startTransactionOnLocationChange: options.instrumentNavigation,
+  });
```

Comment:
To confirm: We can't get rid of setting the hooks globally because we can't really pass them into `withSentry`?

Reply:
I believe we can pass them into `withSentry`. We can try switching to that.

Reply:
cc @AbhiPrasad / @mydea  wdyt would this be a good change? 

My PoV:
- cleaner as we don't need to set anything globally
- but requires more changes for users when migrating (also more wizard changes for us).